pkg/controller/chaosengine/chaosengine_controller.go

```diff
@@ -252,21 +252,25 @@ func initializeApplicationInfo(instance *litmuschaosv1alpha1.ChaosEngine, appInf
 	if instance == nil {
 		return nil, errors.New("empty chaosengine")
 	}
-	appLabel := strings.Split(instance.Spec.Appinfo.Applabel, "=")
-	chaosTypes.AppLabelKey = appLabel[0]
-	chaosTypes.AppLabelValue = appLabel[1]
-	appInfo.Label = make(map[string]string)
-	appInfo.Label[chaosTypes.AppLabelKey] = chaosTypes.AppLabelValue
 
-	if instance.Spec.Appinfo.Appns != "" {
-		appInfo.Namespace = instance.Spec.Appinfo.Appns
-	} else {
-		appInfo.Namespace = instance.Namespace
+	if instance.Spec.AnnotationCheck == "true" {
```

Comment:
Even w/o annotation check set to true, we still have value for applabel and appns -- these are passed to the experiment via runner anyway.

Having said that the split of label should be attempted only when the label is not empty - which is necessary, yes.